include/boost/config/compiler/nvcc.hpp

```diff
@@ -11,6 +11,15 @@
 #  define BOOST_COMPILER "NVIDIA CUDA C++ Compiler"
 #endif
 
+#if defined(__CUDACC_VER_MAJOR__) && defined(__CUDACC_VER_MINOR__) && defined(__CUDACC_VER_BUILD__)
+#  define BOOST_CUDA_VERSION __CUDACC_VER_MAJOR__ * 10000 + __CUDACC_VER_MINOR__ * 100 + __CUDACC_VER_BUILD__
+#elif defined(__CUDACC_VER__)
```

Comment:
note: at least in CUDA 9 RC2 this var will be defined and `__CUDACC_VER__` *is* set to a string with an error message.

since the other 3 are defined above, this `elif` should not be taken but it's maybe still a bit risky.